### PR TITLE
[Issue 2583] opportunity detail error page

### LINKED ISSUE #2583: Direct to error page instead of not found on opportunity detail 500 errors
### Summary

Currently we direct all errors on opportunity listing API calls to a 404 not found page. In order to give users a better sense of the situation we should direct to an error page in non 404 situations.

### Acceptance criteria

- [ ] when a user visits an opportunity detail page (/opportunity/[id]) and the opportunity listings API endpoint ("GET" /opportunities/[id]) returns an error with a 500 status code, the user should be directed to a generic error page rather than a not found or 404 page

### PR DESCRIPTION
**Pull request by doug-s-nava**
## Summary
Fixes #2583

## Changes proposed
Added a generic error page for the opportunity detail route and adjusted logic to display it in non 404 error situations.

### Test steps

1. visit an opportunity detail page for an opportunity that doesn't exist, ex: http://localhost:3000/es/opportunity/44444444
2. _VERIFY_: the 404 / not found page is displayed
3. update the API such that it sends a 500 error. Ex. update src/api/opportunities_v1/opportunity_routes.py line 198:
```
opportunity = get_opportunity(db_session, "show me an error")
```
4. restart backend containers. Run `make stop && make build && make start` in the API directory
5. visit an opportunity detail page for an opportunity that doesn't exist, ex: http://localhost:3000/es/opportunity/4
6. _VERIFY_: a generic error page is shown

## Additional information

Screenshot 2024-10-25 at 2 25 10 PM

### CODE REVIEW

File: frontend/src/app/[locale]/opportunity/[id]/error.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}: {
+  error: Error & { digest?: string };
+}) {
+  useEffect(() => {
+    console.error(error);
+  }, [error]);
+  return (
+    <>
+      <BetaAlert />
+      <SearchErrorAlert />
```
Comment (acouch) [inline frontend/src/app/[locale]/opportunity/[id]/error.tsx:19]: Should we rename this to a more generic error?

**Review by acouch — DISMISSED** [review]
This is a great catch!

File: frontend/src/errors.ts [hunk 2]
```diff
@@ -4,6 +4,17 @@
 
 import { QueryParamData } from "src/services/search/searchfetcher/SearchFetcher";
 
+export const parseErrorStatus = (error: ApiRequestError): number => {
```
Comment (doug-s-nava) [inline frontend/src/errors.ts:7]: note that I don't like that we have to do this, and wonder if there's a better way to pass our errors around that doesn't involve serializing and deserializing JSON

**Review by acouch — APPROVED** [review]
Updates look 🦅 (awesome)

**Pull request merged**